### PR TITLE
BACKLOG-12718: handle default values for extends mixin in CE edit

### PR DESCRIPTION
**Pull request by jkevan**

### CODE REVIEW

File: src/javascript/EditPanel/EditPanel.utils.js [hunk 1]
```diff
@@ -36,15 +36,16 @@ export function getDynamicFieldSets(sections) {
...
  */
-export function getFields(sections, sectionName) {
+export function getFields(sections, sectionName, fieldSetFilter) {
     return sections.reduce((result, section) => {
         let fields = [];
 
         if (!sectionName || sectionName === section.name) {
             fields = section
-                .fieldSets
+                .fieldSets.filter(fieldset => fieldSetFilter ? fieldSetFilter(fieldset) : true)
```
Comment (FBerthelot) [inline src/javascript/EditPanel/EditPanel.utils.js]:
```suggestion
                .fieldSets
                .filter(fieldset => fieldSetFilter ? fieldSetFilter(fieldset) : true)
```

File: src/javascript/FormDefinitions/FormData.adapter.js [hunk 2]
```diff
@@ -17,3 +19,29 @@ export const adaptSections = sections => {
         }, [])
         .filter(section => (section.fieldSets && section.fieldSets.length > 0));
 };
+
+export const getFieldValuesFromDefaultValues = field => {
```
Comment (FBerthelot) [inline src/javascript/FormDefinitions/FormData.adapter.js]: no test on this function ?

File: src/javascript/Edit/Edit.adapter.js [hunk 3]
```diff
@@ -1,20 +1,24 @@
...
     // Retrieve dynamic fieldSets
     const dynamicFieldSets = getDynamicFieldSets(sections);
 
     // Retrieve fields and the return object contains the field name as the key and the field value as the value
-    const fields = getFields(sections)
+    const nodeValues = getFields(sections)
         .reduce((result, field) => ({...result, ...getFieldValues(field, nodeData)}), {});
 
+    // Get default values for not enabled mixins
+    const extendsMixinFieldsDefaultValues = getFields(sections, undefined, fieldset => fieldset.dynamic && !fieldset.activated)
```
Comment (FBerthelot) [inline src/javascript/Edit/Edit.adapter.js]:
```suggestion
    const extendsMixinFieldsDefaultValues = getFields(sections)
```
getFields doesn't accept a function as a third parameter

Comment (FBerthelot) [reply src/javascript/Edit/Edit.adapter.js]: maybe you wanted a filter ?

File: src/javascript/FormDefinitions/FormData.adapter.js (hunk 2 above)
Comment (jkevan) [reply src/javascript/FormDefinitions/FormData.adapter.js]: it's already tested by higher call on other fct like adaptEditData or adaptCreateData, spent already a lot of time on tests, so I didnt want to extract the logics of this tests to test this extracted function

File: src/javascript/Edit/Edit.adapter.js (hunk 3 above)
Comment (jkevan) [reply src/javascript/Edit/Edit.adapter.js]: getFields does accept a 3rnd params that is a fct: 
 * @param {array} sections    array object contains sections
 * @param {string} sectionName string value refer to the section name	 * @param {string} sectionName string value refer to the section name
 * @param {string} fieldSetFilter optional fieldset filter

fieldSetFilter

**Pull request merged**